src/lib.rs

```diff
@@ -82,7 +82,7 @@
 //!
 //! [faq]: https://github.com/rayon-rs/rayon/blob/master/FAQ.md
 
-#[cfg(test)]
+#[cfg(doctest)]
```

Comment:
This seems to cause compile to fail on Rust 1.38 and Rust 1.39:

```console
$ cargo +1.39.0 check
error[E0658]: `cfg(doctest)` is experimental and subject to change
  --> src/lib.rs:85:7
   |
85 | #[cfg(doctest)]
   |       ^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/62210
```

Reply:
It's been implemented in the 1.40 version.

Reply:
Hmm, it would be OK if it were just ineffective before 1.40, but an error is actively harmful. I don't see a lint name to `allow` for this, so I think we'll have to revert it for now.

Alternatively, maybe we can just move this to `tests/`? Then it wouldn't affect the compilation of the main library, and we only run tests in CI on stable/beta/nightly.

Reply:
Absolutely! Do you want me to send a PR to fix it?

Reply:
If you can, that would be great, thanks!

Reply:
I just thought: if moved in another file, then we'll have to run `rustdoc --test` "by hand" (because `rustdoc` isn't run on the `tests` folder. And my knowledge of github-actions is non-existent. So basically, someone needs to copy the two lines of `doctest` into a file on its own and then run `rustdoc --test` on it.

Reply:
Is there any downside to going back to `cfg(test)`?

Reply:
It doesn't get tested since only `rustdoc --test` is testing it, and `cfg(test)` isn't set, preventing `rustdoc` to "see" this and to parse the readme file.

Reply:
So it was never doing anything before this PR? Or did `rustdoc` formerly set `cfg(test)`?

Reply:
It did for a few months then we changed it (and this is why I sent the PR in the first place). To test it, add a failing example in the readme then run `cargo test --doc`. :)

Reply:
I moved it to rayon-demo in #763.